From d01ba15887d0cec23075601a5a239fed5f277332 Mon Sep 17 00:00:00 2001
From: Alex Leibovich <alexl@marvell.com>
Date: Wed, 1 May 2019 17:01:39 +0300
Subject: [PATCH 199/386] mmc: dts: cn9130: fix mmc high speed

Added missing ap_mmc regulator and defined
relevant GPIO pin

Change-Id: Ib11d7d28c784e77cdc76c5e5f6af0f222f92e63c
Signed-off-by: Alex Leibovich <alexl@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/8356
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130-db-A.dts |  1 +
 arch/arm64/boot/dts/marvell/cn9130-db-B.dts |  1 +
 arch/arm64/boot/dts/marvell/cn9130-db.dtsi  | 10 ++++++++++
 3 files changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/cn9130-db-A.dts b/arch/arm64/boot/dts/marvell/cn9130-db-A.dts
index b2b91da545f8..a40d55e2901e 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/cn9130-db-A.dts
@@ -38,6 +38,7 @@
 	pinctrl-names = "default";
 	bus-width = <8>;
 	status = "okay";
+	vqmmc-supply = <&ap0_reg_sd_vccq>;
 };
 
 /*
diff --git a/arch/arm64/boot/dts/marvell/cn9130-db-B.dts b/arch/arm64/boot/dts/marvell/cn9130-db-B.dts
index f6b7b3c99624..32c46e3122c7 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/cn9130-db-B.dts
@@ -37,6 +37,7 @@
 	pinctrl-names = "default";
 	bus-width = <8>;
 	status = "okay";
+	vqmmc-supply = <&ap0_reg_sd_vccq>;
 };
 
 /*
diff --git a/arch/arm64/boot/dts/marvell/cn9130-db.dtsi b/arch/arm64/boot/dts/marvell/cn9130-db.dtsi
index 003b1b364d5f..ef5d58aa23e4 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130-db.dtsi
@@ -49,6 +49,16 @@
 			  3300000 0x0>;
 	};
 
+	ap0_reg_sd_vccq: ap0_sd_vccq@0 {
+		compatible = "regulator-gpio";
+		regulator-name = "ap0_sd_vccq";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+		gpios = <&expander0 8 GPIO_ACTIVE_HIGH>;
+		states = <1800000 0x1
+			  3300000 0x0>;
+	};
+
 	cp0_reg_sd_vcc: cp0_sd_vcc@0 {
 		compatible = "regulator-fixed";
 		regulator-name = "cp0_sd_vcc";
-- 
2.17.1

